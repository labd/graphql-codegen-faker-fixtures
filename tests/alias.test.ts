import { parse } from "graphql";
import { buildSchema } from "graphql/utilities";

import { plugin } from "../dist/index";

const schema = /* GraphQL */ `
  type Person {
    name: String
    age: String
    address: String
  }
`;

const document = /* GraphQL */ `
  fragment person on Person {
    aliasedName: name
    age
  }
`;

describe("alias", () => {
  it("should create a builder using aliased fields", () => {
    const response = plugin(
      buildSchema(schema),
      [{ document: parse(document) }],
      { buildersOnly: true },
    );
    expect(response).toMatchSnapshot();
    expect(response).toContain(
      "aliasedName: faker.lorem.words(),age: faker.lorem.words()",
    );
  });
});
