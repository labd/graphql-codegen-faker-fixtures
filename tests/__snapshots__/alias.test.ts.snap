// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alias should create a builder using aliased fields 1`] = `
"
    export const fakePerson = (override?:DeepPartial<types.PersonFragment>):types.PersonFragment => {
      const fixture:types.PersonFragment = {
        aliasedName: faker.lorem.words(),age: faker.lorem.words()
      }

      return override
        ? merge(fixture, override)
        : fixture
    }
  "
`;
